pkg/apis/core/v1alpha1/build_types.go

```diff
@@ -19,6 +19,8 @@ type BuildBuilderSpec struct {
 	// +patchStrategy=merge
 	// +listType
 	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,15,rep,name=imagePullSecrets"`
+	Kind             string                        `json:"kind,omitempty"`
```

Comment:
I think this may unnecessarily confuse the Build Spec by implying that these fields have behavior. We already print info [logging from an an annotation](https://github.com/pivotal/kpack/blob/cecb3373638e6cc440bf7ae5d74c459987cfacad/pkg/apis/build/v1alpha2/build_pod.go#L382) can we use a similar strategy here?